api/auth/client.go

```diff
@@ -18,171 +18,157 @@ type Auth struct {
 	api restapi.Connector
 }
 
-// New creates a new auth client instance, using the
-// argument SDK API client.
+// New auth client constructor.
 func New(api restapi.Connector) *Auth {
 	return &Auth{api: api}
 }
 
-// AuthStatus get microservice status
-func (store *Auth) AuthStatus() (*common.ServiceStatus, error) {
-	status := &common.ServiceStatus{}
+// MARK: Status
+// Status get auth microservice status.
+func (c *Auth) Status() (*response.ServiceStatus, error) {
+	status := &response.ServiceStatus{}
 
-	_, err := store.api.
+	_, err := c.api.
 		URL("/auth/api/v1/status").
 		Get(status)
 
 	return status, err
 }
 
+// MARK: Identity Provider
 // CreateIdpClient creates a new identity provider client configuration.
-func (store *Auth) CreateIdpClient(idpClient *IDPClient) (IDstruct, error) {
-	idpClientId := IDstruct{}
+func (c *Auth) CreateIdpClient(idpClient *IdpClient) (*response.Id, error) {
```

Comment:
Not sure if returning a pointer is necessary. Can we just return a plain struct? Also, the name response.Id is a bit misleading. Can it be Identifier, or ResourceId?